src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig
+
+trait ConfigFieldMapping[T] {
```

Comment:
I think I can make the necessary changes for the new proposal today. I think it's OK to release a new version now and have the configuration system included in the next version, though.